ocrd_network/ocrd_network/processing_server.py

```diff
@@ -670,3 +700,116 @@ async def list_processors(self) -> List[str]:
             unique_only=True
         )
         return processor_names_list
+
+    async def task_sequence_to_processing_jobs(
+            self,
+            tasks: List[ProcessorTask],
+            mets_path: str,
+            page_id: str,
+            agent_type: str = 'worker',
+    ) -> List[PYJobOutput]:
+        file_group_cache = {}
+        responses = []
+        for task in tasks:
+            # Find dependent jobs of the current task
+            dependent_jobs = []
+            for input_file_grp in task.input_file_grps:
+                if input_file_grp in file_group_cache:
+                    dependent_jobs.append(file_group_cache[input_file_grp])
+            # NOTE: The `task.mets_path` and `task.page_id` is not utilized in low level
+            # Thus, setting these two flags in the ocrd process workflow file has no effect
+            job_input_data = PYJobInput(
+                processor_name=task.executable,
+                path_to_mets=mets_path,
+                input_file_grps=task.input_file_grps,
+                output_file_grps=task.output_file_grps,
+                page_id=page_id,
+                parameters=task.parameters,
+                agent_type=agent_type,
+                depends_on=dependent_jobs,
+            )
+            response = await self.push_processor_job(
+                processor_name=job_input_data.processor_name,
+                data=job_input_data
+            )
+            for file_group in task.output_file_grps:
+                file_group_cache[file_group] = response.job_id
+            responses.append(response)
+        return responses
```

Comment:
Looks good, that should be general enough to handle even dependencies of non-contiguous jobs AFAICT.

Reply:
Yes, that's the case.